### DISCUSSION TITLE
How can I configure the list of cached resources at build time?

### DISCUSSION THREAD

**Discussion by dartess** [vercel/community#7299] [Feature Question] [answered]
### Question

https://vercel.com/docs/deployments/troubleshoot-a-build#understanding-build-cache says:

> Vercel caches files based on the Framework Preset selected in your Project settings. The following files are cached in addition to node_modules/**, yarn.lock, and package-lock.json.
> ...
> Next.js	.next/cache/**

So, if I have the Framework Preset `Next.js` specified in my Build & Development Settings, I get the following list for caching:

* `node_modules/**`
* `yarn.lock`
* `package-lock.json`
* `.next/cache/**`

Also in the same documentation there are several ways to _completely_ disable the build cache during build.

But, can I _partially_ disable cached items? For example, disable cache only of `node_modules/**`, or of `.next/cache/**`?

In other words, can I configure the list to cache resources myself?

**Comment by pawlean** [comment, marked as answer]
Hi @dartess!

Unfortunately, it's currently not possible to partially disable cached items or customize the list of [cached resources](https://vercel.com/docs/deployments/troubleshoot-a-build#understanding-build-cache) based on the Framework Preset in Vercel. 

I'd love to understand what you're trying to do here. What is the benefit that you see of partially disabling cached items?

**Comment by dartess** [comment]
From time to time (rarely, but still) we get a broken build for an unknown reason, and rebuilding with the cache disabled fixes.

The last time a cache problem concerned a third-party plugin, which did not coordinate its behaviour with the Next caching mechanism. This time it is not a problem with Next itself, but we were forced to disable caching.

So I was looking for an option to disable `.next/cache/**` caching but save `node_modules/**` caching. But as it turned out, the build time with the cache completely disabled increased only slightly, so my request is not relevant.

**Reply by pawlean** [reply]
If it’s a rare issue, you can simply rebuild without the cache, and it will update for future builds. However, if you prefer to disable it entirely, that works as well.

Thanks for coming back to this, @dartess! Helps a lot.

[a comment by an automated account is not shown]
